### ISSUE TITLE
TensorFlow or Keras version

### ISSUE THREAD

**Issue by ntelo007** [dmitrysarov/clDice#1]
Hi,

could you help me create a Keras or TensorFlow version of this loss function?

**Comment by jacobkoenig** [comment]
@ntelo007 try this [repository](https://github.com/jacobkoenig/clDice-Loss) where I converted this code to work with Keras/Tensorflow

**Comment by ntelo007** [comment]
Thank you. Have you tested this loss function? Did you retrieve comparable results with the paper? Does it work?

**Comment by dmitrysarov** [comment]
on an experiment with noisy labels (labels came from something like Frangi filter + manual filtration), I retrieve sensible gain 
![image](https://user-images.githubusercontent.com/19648595/80191944-1d2fc700-861f-11ea-87a8-f051793f164f.png)
On clean dataset usage of dice+cldice result in minor performance gain expressed in variance reduction
![image](https://user-images.githubusercontent.com/19648595/80193732-063ea400-8622-11ea-8981-f0663121f690.png)
Please share your experience

**Comment by ntelo007** [comment]
Unfortunately, it is probably not working. I tried training a road detection algorithm and these are the results of the training procedure (both for only clDice and with the combination with dice):

Only with the clDice loss:

![image](https://user-images.githubusercontent.com/31326655/80292179-c5937780-8754-11ea-9ecf-39594a3ef054.png)
![image](https://user-images.githubusercontent.com/31326655/80292182-cf1cdf80-8754-11ea-9885-99a35433ccb4.png)


clDice + dice:
![image](https://user-images.githubusercontent.com/31326655/80292193-ed82db00-8754-11ea-9d68-27794a21a10a.png)
![image](https://user-images.githubusercontent.com/31326655/80292199-f70c4300-8754-11ea-87e3-7d755bdf913d.png)

Any ideas why this is happening?

**Comment by dmitrysarov** [comment]
it looks like the whole training procedure is wrong. Try first to overfit your model on a small dataset (it can be several images, but not one if you use batchnorm).  If the model is able to remember small dataset labels, then move to the evaluation of the validation procedure.

**Comment by ntelo007** [comment]
The training procedure works well for many other loss functins. This means that the problem occurs only when I use the soft_clDice_loss function. This is the prediction result:

![clDice_same_problem](https://user-images.githubusercontent.com/31326655/80910672-6ecc0600-8d31-11ea-9a94-7440db1b201c.png)

It's obvious that the soft_skeletonization procedure is not correct. Because I don't think the math behind the comparison is the problem.

**Comment by dmitrysarov** [comment]
@ntelo007 have you look by eyes on the train image after soft_skeletonization? How is model loss log curves look like when you train only on dice?

**Comment by dmitrysarov** [comment]
in my experiments, cldice makes prediction prone to the thickness rather thinness
